tests/utils/test_request_utils.py

```diff
@@ -0,0 +1,32 @@
+import os
+import subprocess
+
+
+def test_request_utils_does_not_import_mlflow():
+    file_content = """
+import importlib.util
+import os
+import sys
+
+file_path = os.path.join(os.path.dirname(__file__), "../../mlflow/utils/request_utils.py")
+module_name = "mlflow.utils.request_utils"
+
+spec = importlib.util.spec_from_file_location(module_name, file_path)
+module = importlib.util.module_from_spec(spec)
+sys.modules[module_name] = module
+spec.loader.exec_module(module)
+
+assert "mlflow" not in sys.modules
+assert "mlflow.utils.request_utils" in sys.modules
+"""
+
+    test_file_name = os.path.join(
+        os.path.dirname(__file__), "test_request_utils_does_not_import_mlflow.py"
+    )
+
+    try:
+        with open(test_file_name, "w") as f:
```

Comment:
When creating a file or directory, we should use the `tmp_path` fixture to avoid leaving zombie resources on the machine: https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

Reply:
A directory created by `tmp_path` is removed when a test completes. We don't need try-catch.

Reply:
I removed the file finally. I think tmp_path create a random path, but the file_path 'file_path = os.path.join(os.path.dirname(__file__), "../../mlflow/utils/request_utils.py")' depends on the absolute path of request_utils.py file, so it might be hard to get its path from a random generated file path?

Reply:
I think https://github.com/mlflow/mlflow/pull/8352#discussion_r1206170635 can solve that issue.